ratis-server/src/main/java/org/apache/ratis/server/leader/LogAppenderBase.java

```diff
@@ -158,6 +158,11 @@ private TermIndex getPrevious(long nextIndex) {
       }
     }
 
+    final long raftLogLastPurgedIndex = getRaftLog().getStartIndex() - 1;
+    if (raftLogLastPurgedIndex == previousIndex) {
```

Comment:
Let's replace the if-condition as below so that we can remove `raftLogLastPurgedIndex`.
```java
    if (getRaftLog().getStartIndex() == nextIndex) {
```